Tools/Taxonomy/Cwe/CweTaxonomyGenerator.cs

```diff
@@ -0,0 +1,331 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using CsvHelper;
+using Microsoft.CodeAnalysis.Sarif;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Serialization;
+using Taxonomy.Common;
+
+namespace Taxonomy.Cwe
+{
+    public class CweTaxonomyGenerator : TaxonomyGenerator
+    {
+        public List<CweCsvRecord> ReadFromCsv(string filePath)
+        {
+            using FileStream input = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
+            using var textReader = new StreamReader(input);
+            using var csvReader = new CsvReader(textReader, CultureInfo.InvariantCulture);
+
+            return csvReader.GetRecords<CweCsvRecord>().ToList();
+        }
+
+        public Weakness_Catalog ReadFromXml(string filePath)
+        {
+            XmlSerializer serializer = new XmlSerializer(typeof(Weakness_Catalog));
+            return (Weakness_Catalog)serializer.Deserialize(new XmlTextReader(filePath));
+        }
+
+        public void SaveCsvToSarif(string filePath)
```

Comment:
>SaveCsvToSarif [](http://example.com/codeflow?start=20&length=14)

maybe, we should run this inside a try/catch, return bool if succeeded and use that in the program.cs

Reply:
done